### PR TITLE
Sync: adapt checkout endpoint for immediate full sync

### PR DESCRIPTION
**Pull request by lezama**
This will make the `pull` command work again with D36239

### Testing instructions

* See D36239

### CODE REVIEW

File: json-endpoints/jetpack/class.jetpack-json-api-sync-endpoint.php [hunk 1]
```diff
@@ -48,7 +48,7 @@ protected function validate_queue( $query ) {
 			return new WP_Error( 'invalid_queue', 'Queue name is required', 400 );
 		}
 
-		if ( ! in_array( $query, array( 'sync', 'full_sync' ) ) ) {
+		if ( ! in_array( $query, array( 'sync', 'full_sync', 'immediate' ) ) ) {
 			return new WP_Error( 'invalid_queue', 'Queue name should be sync or full_sync', 400 );
```
Comment (gravityrail) [inline json-endpoints/jetpack/class.jetpack-json-api-sync-endpoint.php]: You should update the error message here too

File: json-endpoints/jetpack/class.jetpack-json-api-sync-endpoint.php [hunk 2]
```diff
@@ -233,6 +241,32 @@ protected function result() {
...
+		foreach ( func_get_args()[0] as $key => $item ) {
+			$items[ $key ] = $item;
+		}
+	}
+
+	public function immediate_full_sync_pull( $number_of_items ) {
+		// try to give ourselves as much time as possible.
+		set_time_limit( 0 );
+
+		remove_filter( 'jetpack_sync_send_data', array( 'Automattic\Jetpack\Sync\Actions', 'send_data' ), 10, 6 );
```
Comment (gravityrail) [inline json-endpoints/jetpack/class.jetpack-json-api-sync-endpoint.php]: `remove_filter` doesn't support the "number of args" param, set here to 6. You can leave that out. Also the default for the third arg is 10, so you can leave that out too.

You might also want to reduce typos and make the code more readable by making the callbacks a var, like this:

```php
$original_send_data_cb = [ 'Automattic\Jetpack\Sync\Actions', 'send_data' ];
$temp_send_data_cb = [ $this, 'jetpack_sync_send_data_listener' ];

remove_filter( 'jetpack_sync_send_data', $original_send_data_cb );
add_filter( 'jetpack_sync_send_data', $temp_send_data_cb, 10, 6 );

Sender::get_instance()->do_full_sync();

remove_filter( 'jetpack_sync_send_data', $temp_send_data_cb );
add_filter( 'jetpack_sync_send_data', $original_send_data_cb, 10, 6 );
```

**Review by gravityrail — CHANGES_REQUESTED** [review]
Some minor suggestions

File: json-endpoints/jetpack/class.jetpack-json-api-sync-endpoint.php [hunk 3]
```diff
@@ -233,6 +241,32 @@ protected function result() {
 		);
 	}
 
+	public $items = [];
+
+	public function jetpack_sync_send_data_listener() {
+		foreach ( func_get_args()[0] as $key => $item ) {
```
Comment (enejb) [inline json-endpoints/jetpack/class.jetpack-json-api-sync-endpoint.php]: I don't understand what this does. Since this function doesn't return anything...

**Comment by enejb** [comment]
I have tested this PR with the .com cli command it thing seemed to have worked as expected. 

I am not sure if it because of the filter that I used?

File: json-endpoints/jetpack/class.jetpack-json-api-sync-endpoint.php (hunk 3 above)
Comment (lezama) [reply json-endpoints/jetpack/class.jetpack-json-api-sync-endpoint.php]: it stores actions into `$this->items` that we then send.

**Comment by jeherve** [comment]
Noting that this will need to be cherry-picked to `branch-8.sync` when merged. I'll take care of that.

**Review by gravityrail — APPROVED** [review]
LGTM!

File: json-endpoints/jetpack/class.jetpack-json-api-sync-endpoint.php (hunk 3 above)
Comment (enejb) [reply json-endpoints/jetpack/class.jetpack-json-api-sync-endpoint.php]: 🤦‍♂

**Comment by enejb** [comment]
Nice work

**Comment by lezama** [comment]
@jeherve, @dereksmart not sure how to bypass codeclimate, it would be amazing if we could add this to `branch-8.sync` as soon as possible. Thanks!

**Review by kraftbj — APPROVED** [review]

**Comment by kraftbj** [comment]
Merged and cherry-picked to `branch-8.sync` via `aacd55987` @lezama

**Pull request merged**